### DISCUSSION TITLE
Viewing and Managing Individual Tasmota Devices in Amazon Web Services IOT

### DISCUSSION THREAD

**Discussion by lfjm2022** [arendst/Tasmota#18030] [Q&A]
I have just started experimenting with tasmota and AWS IOT. I successfully worked my way through the [Tasmota walkthrough](https://tasmota.github.io/docs/AWS-IoT/) and have connected a device to AWS and used the MQTT test functionality in AWS IOT to confirm it is connected and sending messages like it shows in the tutorial. 

I would like to be able to monitor data and have access control for each device I connect such as revoking its connection to the server. Right now I can not see what devices are connected unless I listen for the wildcard (#) topic and monitor all the data coming in and tasmota does not seem to support AWS's version of device tracking and authorization called "Shadows" which would show me the individual devices, their MQTT messages being sent, and - it appears - allow me to revoke the device access to the server. Does anyone have experience with connecting tasmota to shadows or something similar for device management that could recommend a route forward? Thanks!

**Comment by s-hadinger** [comment]
`SetOption84 1` enables AWS IoT shadow support.

**Reply by lfjm2022** [reply]
Thanks! I tried that with my device but I am not seeing the ```xx:xx:xx MQT: Updated shadow: $aws/things/[device_name]/shadow/update``` in my Tasmota WebConsole. I am also not seeing any shadows beyond ones I have already generated.

Do I need to have a shadow premade for my device or will tasmota and AWS automatically generate a shadow when I pass `SetOption84 1`?

**Comment by thermseekr** [comment]
Hi @lfjm2022 , I too want to link AWS IoT Things to my tasmota devices. Have you found out how to do it?

**Reply by s-hadinger** [reply]
Have you looked here?
https://tasmota.github.io/docs/AWS-IoT/

What's missing?

**Reply by thermseekr** [reply]
Yes I've looked, I even sent a PR today to update some parts of that doc. It works perfectly with the IoT Core's MQTT test client. I meant using things/shadows to monitor the device. @lfjm2022  asked a question about SetOption 84 that was not answered, I was wondering if he figured out how to use shadows with tasmota devices. I'm currently using the user/password authentication, should I change to using key pairs and certificates to achieve that?

**Comment by matriedel** [comment]
I am also evaluating how to subscribe to AWS IoT Things. 


My test setup is a nodemcu ESP8266EX with LED ring/neopixel and a BH1750. 
Tasmota Version 12.4.0.4(tasmota)-2_7_4_9(2023-04-15T15:47:49)

To publish a message from AWS to the nodemcu works fine when using the command:

cmnd/tasmota_80DC64/json

{"POWER":"ON","Dimmer":100,"Color":"06FF00","HSBColor":"119,100,100","Channel":[2,100,0]}  

As a result, I see the LEDs changing colour.

Next step. Once confirmed that communication is possible I tried with a rule to subscribe to an AWS IoT device sending colour Mqtt messages. with 

Rule1
  ON mqtt#connected DO Subscribe TestLED, $aws/things/XXXXXXXXXX/shadow/get/accepted ENDON
  ON Event#TestLED DO SerialLog "Color %value%” ENDON

But I do not receive anything. Any hint about what might be the issue?


I also checked with

Rule1
ON mqtt#connected DO SerialLog "MQTT connected" ENDON

to evaluate if mqtt#connected results in a trigger once connected, but this is not the case

**Reply by s-hadinger** [reply]
You should try commands individually first. `SerialLog` doesn't output anything in the logs, its sole purpose is to set the log level.

Also set `SerialLog 4` or `weblog 4` so you will see all low-level traffic in the logs.

Did you subscribe to `$aws/things/XXXXXXXXXX/shadow/get/accepted` ? If you don't subscribe, AWS IoT does not send it to you.

**Reply by matriedel** [reply]
Yes, I subscribed to $aws/things/XXXXXXXXXX/shadow/get/accepted using this rule Rule1. Do you think this is not how you can realize a valid subscription?

Rule1
ON mqtt#connected DO Subscribe TestLED, $aws/things/XXXXXXXXXX/shadow/get/accepted ENDON
ON Event#TestLED DO SerialLog "Color %value%” ENDON


What I found is it seems "It seems ON mqtt#connected DO" is not triggered

If I run "Subscribe TestLED, $aws/things/XXXXXXXXXX/shadow/get/accepted ENDON"  alone via the command line I get at least

11:09:51.424 MQT: Subscribe to $aws/things/XXXXXXXXXX/shadow/get/accepted ENDON/#
11:09:51.489 MQT: stat/tasmota_80DC64/RESULT = {"Subscribe":"TESTLED,$aws/things/XXXXXXXXXX/shadow/get/accepted ENDON/#"}
